src/MessagingInput/MessagingInput.tsx

```diff
@@ -6,40 +6,42 @@ import KeyCode from "../utils/KeyCode";
 import * as FontAwesome from "react-fontawesome";
 
 import "./MessagingInput.less";
+import { useRef, useImperativeHandle } from "react";
 
 function cssClass(element: string) {
   return `MessagingInput--${element}`;
 }
 
 interface Props {
+  attachments?: React.ReactNode[];
   className?: string;
-  newlineOnEnter?: boolean;
-  value: string;
+  checkbox?: {
+    isChecked: boolean;
+    isVisible: boolean;
+    label: React.ReactNode;
+    onChange: (value: boolean) => void;
+  };
+  disableSendButton?: boolean;
+  /** Temporarily added to allow overriding the text with a translation. */
+  labelText?: string;
+  onBlur?: () => void;
   onChange: (newValue: string) => void;
+  onFocus?: () => void;
```

Comment:
This is the only new prop - the rest are just alphabetized but would appreciate a sanity check! With all the props now passed in, I thought that the component would benefit from some organization